code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm

```diff
@@ -110,8 +111,10 @@
 
 /obj/machinery/atmospherics/components/unary/thermomachine/process_atmos()
 	..()
-	if(!on || !nodes[1])
+	if(!(is_operational()) || !on || !nodes[1])  //if it has no power or its switched off, dont process atmos
 		return
+	else if(is_operational() && was_on)  //if it was switched on before it turned off due to no power, turn the machine back on
+		on = TRUE
```

Comment:
If it was switched off due to having no power on = FALSE, meaning this line will never run